libraries/Microsoft.Bot.Builder.AI.LuisV3/LuisApplication.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Linq;
+using System.Web;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    /// <summary>
+    /// Data describing a LUIS application.
+    /// </summary>
+    public class LuisApplication
+    {
+        public LuisApplication()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LuisApplication"/> class.
+        /// </summary>
+        /// <param name="applicationId">LUIS application ID.</param>
+        /// <param name="endpointKey">LUIS subscription or endpoint key.</param>
+        /// <param name="endpoint">LUIS endpoint to use like https://westus.api.cognitive.microsoft.com.</param>
+        public LuisApplication(string applicationId, string endpointKey, string endpoint)
+            : this((applicationId, endpointKey, endpoint))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LuisApplication"/> class.
+        /// </summary>
+        /// <param name="applicationEndpoint">LUIS application endpoint.</param>
+        public LuisApplication(string applicationEndpoint)
+            : this(Parse(applicationEndpoint))
+        {
+        }
+
+        private LuisApplication(ValueTuple<string, string, string> props)
+        {
+            var (applicationId, endpointKey, endpoint) = props;
+
+            if (!Guid.TryParse(applicationId, out var _))
+            {
+                throw new ArgumentException($"\"{applicationId}\" is not a valid LUIS application id.");
+            }
+
+            if (!Guid.TryParse(endpointKey, out var subscriptionGuid))
```

Comment:
Isn't this adding an assumption about the subscription key? What if Cognitive services decide to make the key with the same format as the storage account's key? Something like this `Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==`.

Reply:
Do you think it likely they will break every cognitive services key by moving away from GUIDs?  If I remove this assumption, there is no early error checking I can do.

Reply:
They could do it the same way they introduced the custom domains. You could always check that it's not null.